### PR TITLE
Glob includes

### PR DESCRIPTION
**Pull request by postwait**

### CODE REVIEW

File: src/mtev_conf.c [hunk 1]
```diff
@@ -435,13 +437,35 @@ mtev_conf_magic_separate_includes(include_node_t **root_include_nodes, int *cnt)
...
-        include_nodes[i].insertion_point->children =
-          include_nodes[i].old_children;
+            n; n = n->next) {
+          if(n->_private == NULL && prev) n->_private = prev->_private;
+          include_node_t *owner = n->_private;
+          assert(owner);
+          n->parent = owner->snippet ? (xmlNodePtr)owner->doc : owner->root;
+          prev = n;
+        }
+        /* unlink the conjuction of lists */
```
Comment (esproul) [inline src/mtev_conf.c]: "conjunction"

File: src/mtev_conf.c [hunk 2]
```diff
@@ -486,18 +510,27 @@ mtev_conf_kansas_city_shuffle_redo(include_node_t *include_nodes, int include_no
...
         else
-          include_nodes[i].insertion_point->children =
-            include_nodes[i].root;
+          more_kids = include_nodes[i].root;
+
+        if(include_nodes[i].glob_idx == 0 || include_nodes[i].insertion_point->children == NULL) {
+          include_nodes[i].insertion_point->children = more_kids;
+        }
+        else {
+          for(n=include_nodes[i].insertion_point->children; n->next; n = n->next);
```
Comment (rileyberton) [inline src/mtev_conf.c]: this loop breaks if `n->next == NULL`?  I don't follow what's happening here.

File: src/mtev_conf.c [hunk 3]
```diff
@@ -945,8 +997,32 @@ mtev_conf_magic_mix(const char *parentfile, xmlDocPtr doc, include_node_t* inc_n
...
-  cnt = xmlXPathNodeSetGetLength(pobj->nodesetval);
-  if(cnt > 0) {
+  nglobs = xmlXPathNodeSetGetLength(pobj->nodesetval);
+  cnt = 0;
+  if(nglobs > 0) {
+    tl_globs = calloc(sizeof(*tl_globs), nglobs);
+    for (i=0; i<nglobs; i++) {
+      node = xmlXPathNodeSetItem(pobj->nodesetval, i);
+      char *path = (char *)xmlGetProp(node, (xmlChar *)"file");
+      char globpat[PATH_MAX];
```
Comment (rileyberton) [inline src/mtev_conf.c]: hoist out of loop?

File: src/mtev_conf.c [hunk 4]
```diff
@@ -958,58 +1034,77 @@ mtev_conf_magic_mix(const char *parentfile, xmlDocPtr doc, include_node_t* inc_n
...
-    else {
-      char *cp;
-      infile = malloc(PATH_MAX);
-      strlcpy(infile, parentfile, PATH_MAX);
-      for(cp = infile + strlen(infile) - 1; cp > infile; cp--) {
-        if(*cp == '/') { *cp = '\0'; break; }
-        else *cp = '\0';
+
+    for (j=0; j<tl_globs[i].gl_pathc; j++) {
+      char infile[PATH_MAX];
```
Comment (rileyberton) [inline src/mtev_conf.c]: hoist?

File: src/mtev_conf.c [hunk 5]
```diff
@@ -958,58 +1034,77 @@ mtev_conf_magic_mix(const char *parentfile, xmlDocPtr doc, include_node_t* inc_n
...
+        char *cp;
+        strlcpy(infile, parentfile, PATH_MAX);
+        for(cp = infile + strlen(infile) - 1; cp > infile; cp--) {
+          if(*cp == '/') { *cp = '\0'; break; }
+          else *cp = '\0';
+        }
+        strlcat(infile, "/", PATH_MAX);
+        strlcat(infile, path, PATH_MAX);
+      }
+mtevL(mtev_debug, "Reading include[%d] file: %s\n", inc_idx, infile);
```
Comment (rileyberton) [inline src/mtev_conf.c]: indent

File: src/mtev_conf.c (hunk 2 above)
Comment (postwait) [inline src/mtev_conf.c]: yup. seeks to the last node.

**Pull request merged**